src/GitTfs/Core/GitTfsRemote.cs

```diff
@@ -1004,13 +1018,27 @@ private IGitTfsRemote InitTfsBranch(RemoteOptions remoteOptions, string tfsRepos
             string sha1RootCommit = null;
             if (rootChangesetId != -1)
             {
-                sha1RootCommit = Repository.FindCommitHashByChangesetId(rootChangesetId);
-                if (fetchParentBranch && string.IsNullOrWhiteSpace(sha1RootCommit))
-                    sha1RootCommit = FindRootRemoteAndFetch(rootChangesetId, renameResult);
-                if (string.IsNullOrWhiteSpace(sha1RootCommit))
-                    return null;
+                try
+                {
+                    sha1RootCommit = Repository.FindCommitHashByChangesetId(rootChangesetId);
+                    if (fetchParentBranch && string.IsNullOrWhiteSpace(sha1RootCommit))
+                        sha1RootCommit = FindRootRemoteAndFetch(rootChangesetId, renameResult);
 
-                Trace.WriteLine("Found commit " + sha1RootCommit + " for changeset :" + rootChangesetId);
+                    if (string.IsNullOrWhiteSpace(sha1RootCommit))
+                        return null;
+
+                    Trace.WriteLine("Found commit " + sha1RootCommit + " for changeset :" + rootChangesetId);
+                }
+                catch (Exception ex)
+                {
+                    // Ignore exceptions "TF14098: Access Denied: User ??? needs
+                    // Read permission(s) for at least one item in changeset ???."
+                    if (!ex.Message.Contains("TF14098"))
+                    {
+                        throw;
+                    }
+                    Trace.WriteLine("No commit found commit changeset " + rootChangesetId + ": " + ex.Message);
```

Comment:
Should this `return null`, since `sha1RootCommit` will be `null`, and it looks like this method returns early if it's unable to set `sha1RootCommit`.

Reply:
Nice catch. Yeah, I will try that and update the PR!